### PR TITLE
Adds support for the Makerdiary nRF52840 Connect Kit board

### PR DESCRIPTION
**Pull request by markbirss**
This PR adds support for the Makerdiary nRF52840 Connect Kit board.

Matches the format of the existing Makerdiary Pitaya Go board

https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/src/boards/pitaya_go/board.h

### CODE REVIEW

File: src/boards/nRF52840_connect_kit/board.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+//--------------------------------------------------------------------+
+#define BLEDIS_MANUFACTURER   "Makerdiary"
+#define BLEDIS_MODEL          "nRF52840 Connect Kit"
+
+//--------------------------------------------------------------------+
+// USB
+//--------------------------------------------------------------------+
+#define USB_DESC_VID            0x2886
+#define USB_DESC_UF2_PID        0xF00E
+#define USB_DESC_CDC_ONLY_PID   0xF00E
```
Comment (hathach) [inline src/boards/nRF52840_connect_kit/board.h:60]: can you provide link where VID/PID is allocated for this board ?

File: src/boards/nRF52840_connect_kit/board.h [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+//--------------------------------------------------------------------+
+// UF2
+//--------------------------------------------------------------------+
+#define UF2_PRODUCT_NAME        "nRF52840 Connect Kit"
+#define UF2_VOLUME_LABEL        "CONNECTKIT"
+#define UF2_BOARD_ID            "CONNECTKIT"
```
Comment (hathach) [inline src/boards/nRF52840_connect_kit/board.h:67]: `UF2_BOARD_ID` does not follow correct format from [uf2 specs](https://github.com/microsoft/uf2#files-exposed-by-bootloaders)

**Review by hathach — CHANGES_REQUESTED** [review]
thanks and please update PR according to review

File: src/boards/nRF52840_connect_kit/board.h (hunk 2 above)
Comment (markbirss) [reply src/boards/nRF52840_connect_kit/board.h]: Ok, ihave updated the UF2_BOARD_ID - Board-ID: nRF52840-Connect-Kit

File: src/boards/nRF52840_connect_kit/board.h (hunk 1 above)
Comment (markbirss) [reply src/boards/nRF52840_connect_kit/board.h:60]: I will wait for makerdiary reply

**Review by markbirss — COMMENTED** [review]
are the changes, ok, i must await makerdiary response?

**Comment by hathach** [comment]
> are the changes, ok, i must await makerdiary response?

thanks, it looks good, though you need to add this board to the ci yml as well https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/.github/workflows/githubci.yml

**Comment by markbirss** [comment]
> > are the changes, ok, i must await makerdiary response?
> 
> thanks, it looks good, though you need to add this board to the ci yml as well https://github.com/adafruit/Adafruit_nRF52_Bootloader/blob/master/.github/workflows/githubci.yml

Sure
https://github.com/adafruit/Adafruit_nRF52_Bootloader/pull/311

**Comment by hathach** [comment]
please push the ci yml changes to this pr

**Comment by markbirss** [comment]
Ok, i  updated githubci.yml

File: .github/workflows/githubci.yml [hunk 3]
```diff
@@ -45,6 +45,7 @@ jobs:
           - 'mikoto'
           - 'nice_nano'
           - 'nrf52840_bboard'
+          - 'nrf52840-connectkit'
```
Comment (hathach) [inline .github/workflows/githubci.yml:48]: must exactly name of board folder

Comment (markbirss) [reply .github/workflows/githubci.yml]: > must exactly name of board folder

Fixed

**Comment by markbirss** [comment]
@hathach 

anything else need to be fixed ?

**Comment by hathach** [comment]
> @hathach
> 
> anything else need to be fixed ?

vid, pid confirmation

**Comment by markbirss** [comment]
@hathach 

OK, will follow up my request i sent to makerdiary

**Comment by markbirss** [comment]
@hathach 

Our partner SeeedStudio has assigned us 2886:F001～2886:F00F, and we give nRF52840-Connect-Kit UF2 Bootloader the 2886:F00F.

https://github.com/makerdiary/nrf52840-connectkit/discussions/2#discussioncomment-6605728

**Comment by markbirss** [comment]
@hathach 

I have updated board.h with the confirmed VID:PID

**Review by hathach — APPROVED** [review]
all is good

**Pull request merged**